enhancements/authentication/unauthenticated-subjects.md

```diff
@@ -0,0 +1,123 @@
+  * "rbac.authorization.k8s.io/autoupdate=true" annotation is also added to default policy resources.
+  * On start-up both openshift-apiserver and kube-apiserver reconcile default policy resources. They do the following operations based on the annotation
+    * if the annotation is not modified.
+        * add the missing permissions to cluster roles or add the missing subjects to cluster role bindings.
+    * if the annotation is deleted
+        * add the annotation back if it is missing
+        * add the missing permissions to cluster roles or add the missing subjects to cluster role bindings.
+    * if the annotation is set to "false"
+        * skips reconciliation of resource
+  * reconciliation does not delete any existing permissions or subjects.
+  * Resources with the annotation set to "false" skips above operations. This allows an admin to opt-out of reconciliation for a specific resource.
+  * Detailed behavior can be found in this [link](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#auto-reconciliation).
+
+* Openshift-apiserver operator implements a new RBACController to reconcile following cluster role bindings created by openshift-apiserver.
+  * cluster-status-binding
+  * discovery
+  * system:openshift:discovery
+  * basic-users
+* If the annotation is not modified, the controller removes the unauthenticated subject from the list of subjects and updates the resource in the API server.
+* Customers can opt-out of this reconciliation by setting the annotation to "false".
+
+
+
+### User Stories [optional]
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigations
+
+Because of the earlier releases of OpenShift allowing access to “/” endpoint, some customer implementations might be expecting this privileged access to anonymous users. Removing unauthenticated subject on an upgrade might break some of these implementations. Informing customers beforehand of this change would help the customer to decide better. Customers can either opt-out of reconciliation or change their implementation to match the new requirements. A customer can opt-out of reconciliation by setting the annotation to "false" before the upgrade.
+
+## Design Details
+
+### Test Plan
+
+Existing e2e test can be re-used to validate the change.
+
+### Graduation Criteria
+
+As this enhancement fixes a bug, graduation criteria is not applicable.
+
+##### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+An alternate implementation is to add an alert to warn of potentially insecure bootstrap policy with instructions to fix. This would not break existing clusters and ensures that they could fix the problem.
```

Comment:
You've mentioned some pros of this alternative. Maybe summarize the cons as well?

Reply:
Added something that I thought is relevant. Please let me know if that is not sufficient.